File: LeetCode/[M] - 1456. Maximum Number of Vowels in a Substring of Given Length/solution.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = set(["a", "e", "i", "o", "u"])
        max_num = 0
        for i in range(k):
            max_num += s[i] in vowels
        window = max_num
        for i in range(k, len(s)):
            window += s[i] in vowels
            window -= s[i - k] in vowels
            max_num = max(max_num, window)
        return max_num

File: LeetCode/[M] - 1456. Maximum Number of Vowels in a Substring of Given Length/solution.go
package main

func maxVowels(s string, k int) int {
	vows := map[byte]struct{}{
		'a': {},
		'e': {},
		'i': {},
		'o': {},
		'u': {},
	}
	mx := 0
	for i := 0; i < k; i++ {
		if _, ok := vows[s[i]]; ok {
			mx++
		}
	}
	wnd := mx
	for i := k; i < len(s); i++ {
		if _, ok := vows[s[i]]; ok {
			wnd++
		}
		if _, ok := vows[s[i-k]]; ok {
			wnd--
		}
		if wnd > mx {
			mx = wnd
		}
	}
	return mx
}
